File: 0454-4sum-ii/0454-4sum-ii.py
class Solution:
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        mp = {}
        res = 0
        for i in nums1:
            for j in nums2:
                mp[-(i+j)] = mp.get(-(i+j), 0) + 1
        for k in nums3:
            for l in nums4:
                res += mp.get(k+l, 0)
        return res

File: 0454-4sum-ii/0454-4sum-ii.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @param {number[]} nums3
 * @param {number[]} nums4
 * @return {number}
 */
var fourSumCount = function(nums1, nums2, nums3, nums4) {
    const map = {}
    let res = 0
    for (const n1 of nums1) {
        for (const n2 of nums2) {
            const sum = n1 + n2
            map[-sum] = map[-sum] + 1 || 1
        }
    }
    for (const n3 of nums3) {
        for (const n4 of nums4) {
            const sum = n3 + n4
            if (sum in map) {
                res += map[sum]
            }
        }
    }
    return res
};